### PR TITLE
Pin descriptions enabled with dialog and colormap

### LINKED ISSUE #425: Show pin description in pin layout
**Expected Behaviour**  
Clicking on a pin in pin layout screen should show what the pin with a small description

**Steps to reproduce it**  
> Not applicable

**Additional context**  
Similar to pin layout in PSLab Android app

**Would you like to work on the issue?**
Issue for GSoC

### PR DESCRIPTION
**Pull request by jddeep**
Fixes #425 

**Changes**: Pin descriptions enabled with dialog and colorMap.

**Screenshot/s for the changes**: 

 

**Checklist**: [Please tick following check boxes with `[x]` if the respective task is completed]
- [x] I have used resources from `strings.xml`, `dimens.xml` and `colors.xml` without hard-coding them
- [x] No modifications done at the end of resource files `strings.xml`, `dimens.xml` or `colors.xml`
- [x] I have reformatted code in every file included in this PR [<kbd>CTRL</kbd>+<kbd>ALT</kbd>+<kbd>L</kbd>]
- [x] My code does not contain any extra lines or extra spaces
- [x] I have requested reviews from other members

**APK for testing**: 
[app-debug.zip](https://github.com/fossasia/neurolab-android/files/3541158/app-debug.zip)

### CODE REVIEW

File: app/src/main/java/io/neurolab/activities/PinLayoutActivity.java [hunk 1]
```diff
@@ -33,16 +46,56 @@ protected void onCreate(Bundle savedInstanceState) {
...
+        populatePinDetails();
+    }
+
+    private void populatePinDetails() {
+        pinDetails.add(new PinDetails("ESP", "ESP Programmer pin", Color.parseColor("#6b40a9"), Color.parseColor("#dc1616")));
+        pinDetails.add(new PinDetails("RXP", "Receiver pin for UART communication", Color.parseColor("#4372a2"), Color.parseColor("#1616dc")));
+        pinDetails.add(new PinDetails("TXP", "Transmitter pin for UART communication", Color.parseColor("#4372a2"), Color.parseColor("#37dc16")));
+        pinDetails.add(new PinDetails("GND", "Ground pin (0 V)", Color.parseColor("#ff4040"), Color.parseColor("#16dcda")));
+        pinDetails.add(new PinDetails("AC1", "Alternative channel input", Color.parseColor("#ffe040"), Color.parseColor("#b01498")));
+        pinDetails.add(new PinDetails("MIC", "External microphone input", Color.parseColor("#7d5840"), Color.parseColor("#890b41")));
```
Comment (CloudyPadmal) [inline app/src/main/java/io/neurolab/activities/PinLayoutActivity.java]: What are these? Did you just copy paste code without checking? These are specific to PSLab

**Review by CloudyPadmal — CHANGES_REQUESTED** [review]

File: app/src/main/java/io/neurolab/activities/PinLayoutActivity.java (hunk 1 above)
Comment (jddeep) [reply app/src/main/java/io/neurolab/activities/PinLayoutActivity.java]: Yes, actually I don't know much about the actual hardware circuit nor have any experience in electronics, so i decided to refer same from the PSLab PIN details for now. @CloudyPadmal from where do you suggest I add these specific pin details? Should I take some help from the hardware repo and update like these : 

Part | References
-- | --
C_Feedthrough | C10 C11 C12 C13
Capacitor | C14 C15 C16 C17 C18 C23 C28 C29 C31 C35 C36 C49 C50 C61 C62
CP | C64 C65 C66 C67 C68 C69
C_Feedthrough | C20 C21
Capacitor | C22 C30 C32 C37 C39 C44
Capacitor | C19
Capacitor | C1 C2 C3 C4 C7 C8 C9 C41 C42 C46 C47 C48 C51 C52 C53 C54 C55 C56 C57 C58 C59 C60
Capacitor | C26 C27 C33 C34
Capacitor | C5 C6 C63
Capacitor | C38 C40 C43 C45

Comment (CloudyPadmal) [reply app/src/main/java/io/neurolab/activities/PinLayoutActivity.java]: Oh my goodness. You know what a **pin** is right?

Comment (CloudyPadmal) [reply app/src/main/java/io/neurolab/activities/PinLayoutActivity.java]: You can see each pin has a small label right next to it in this image right? https://raw.githubusercontent.com/fossasia/neurolab-hardware/master/docs/images/Top.png
Just Google search each name or refer to PSLab app and add that as a small description. Except for AD0~AD3; they are Analog input 0 ~ 3

Comment (jddeep) [reply app/src/main/java/io/neurolab/activities/PinLayoutActivity.java]: Okay. understood. I was referring from the Neurolab_BOM_sheets :sweat_smile: earlier

File: app/src/main/java/io/neurolab/activities/PinLayoutActivity.java [hunk 2]
```diff
@@ -33,16 +46,62 @@ protected void onCreate(Bundle savedInstanceState) {
...
+        pinDetails.add(new PinDetails("C24", "Capacitor 16V 10%", Color.parseColor("#ff4040"), Color.parseColor("#d25c3c")));
+        pinDetails.add(new PinDetails("C25", "Capacitor 16V 10%", Color.parseColor("#4372a2"), Color.parseColor("#5c5c3c")));
+        pinDetails.add(new PinDetails("R7", "Chip resistor pin - 1/16W 20%", Color.parseColor("#7d5840"), Color.parseColor("#41ff0b")));
+        pinDetails.add(new PinDetails("R17", "Chip resistor pin - 1/16W 20%", Color.parseColor("#7d5840"), Color.parseColor("#41ff0b")));
+        pinDetails.add(new PinDetails("R18", "Chip resistor pin - 1/16W 20%", Color.parseColor("#7d5840"), Color.parseColor("#41ff0b")));
+        pinDetails.add(new PinDetails("R25", "Chip resistor pin - 1/16W 20%", Color.parseColor("#7d5840"), Color.parseColor("#41ff0b")));
+        pinDetails.add(new PinDetails("R34", "Chip resistor pin - 1/16W 20%", Color.parseColor("#7d5840"), Color.parseColor("#41ff0b")));
+        pinDetails.add(new PinDetails("R8", "Chip resistor pin - 1/16W 20%", Color.parseColor("#7d5840"), Color.parseColor("#41ff0b")));
```
Comment (CloudyPadmal) [inline app/src/main/java/io/neurolab/activities/PinLayoutActivity.java]: it's "Pin descriptions"! @jddeep do you understand this feature? Did you even try the PSLab Android app?

**Review by CloudyPadmal — CHANGES_REQUESTED** [review]

File: app/src/main/java/io/neurolab/activities/PinLayoutActivity.java (hunk 2 above)
Comment (jddeep) [reply app/src/main/java/io/neurolab/activities/PinLayoutActivity.java]: These are basically resistance measuring pins...so should I simply put `Resistance measurement pin` as description? (similar in PSLab)

Comment (CloudyPadmal) [reply app/src/main/java/io/neurolab/activities/PinLayoutActivity.java]: R8 is a resistor. How do you see it as a **pin**?

**Comment by CloudyPadmal** [comment]
@jddeep if you don't understand what a pin is, I think you can understand what a red circle is. Here I have circled the pins with red. Please try to understand.

Top

File: app/src/main/java/io/neurolab/activities/PinLayoutActivity.java (hunk 2 above)
Comment (CloudyPadmal) [reply app/src/main/java/io/neurolab/activities/PinLayoutActivity.java]: Just try out the PSLab app and see. This issue should not involve such discussion.

**Comment by jddeep** [comment]
> @jddeep if you don't understand what a pin is, I think you can understand what a red circle is. Here I have circled the pins with red. Please try to understand.
> 
> Top

@CloudyPadmal Thanks a lot really. I was really confused with this. But now as I was doing some research, I found out that some were ICs, microcontrollers, resistors, etc. Will update the PR asap.

**Comment by CloudyPadmal** [comment]
@jddeep please try out this feature in PSLab android app as we've said many times in meetings to avoid confusions.

**Review by CloudyPadmal — CHANGES_REQUESTED** [review]
@jddeep the color map is the PSLab one.

**Comment by jddeep** [comment]
@CloudyPadmal Yes, please provide the colormap images for neurolab. I couldn't find the same and hence had to use the PSLab one. Apologies for extending the discussion of this issue.

**Comment by CloudyPadmal** [comment]
Why do I have to provide you the color map? Make it yourself. Look how it's done in PSLab Android app

**Comment by jddeep** [comment]
@CloudyPadmal Updated with correct color maps. Now this works correctly, similar to PSLab feature tested. In the back layout, I have described the I1+, I1-,..pins as current measuring pins. Please do tell me if otherwise, will update the desc. for those.

**Review by CloudyPadmal — CHANGES_REQUESTED** [review]
Refer https://blog.fossasia.org/implementing-clickable-images/

I don't see all the pins in color map and the once are not perfect squares. They are also blurred.

**Comment by adityastic** [comment]
@jddeep here you go mate, use this 👍 

front_pin_colormap

**Comment by adityastic** [comment]
@jddeep here you go mate, use these 👍 

front:

back:

**Comment by jddeep** [comment]
@CloudyPadmal PR updated. Have tested and working fine now. Thank you for the help and patience :smile:

**Comment by adityastic** [comment]
@jddeep no thanks to me 😂😂😂

**Comment by jddeep** [comment]
> @jddeep no thanks to me

@adityastic yes thanks to you as well xD.

**Comment by adityastic** [comment]
>  @adityastic yes thanks to you as well xD.

Haha, I was kidding

**Comment by CloudyPadmal** [comment]
@jddeep update the working branch with dev branch and build the apk to reduce the size.

**Comment by jddeep** [comment]
@CloudyPadmal done.

**Comment by CloudyPadmal** [comment]
@jddeep can you check if this is working for AD0~3? Others work.

**Comment by CloudyPadmal** [comment]
Ah the color map doesn't have them

**Review by CloudyPadmal — CHANGES_REQUESTED** [review]
Doesn't work for AD0 to AD3 pins

**Comment by jddeep** [comment]
@CloudyPadmal Does this colormap seem right with the analog inputs 0~3?

**Comment by adityastic** [comment]
updated with ad0~3
back:
front
front:
front

**Comment by adityastic** [comment]
@jddeep updated with ad0~3

back:

front:

**Comment by CloudyPadmal** [comment]
> @jddeep updated with ad0~3
> 
> back:
> 
> front:
> 

They are all green. How to identify each pin individually?

**Comment by adityastic** [comment]
@CloudyPadmal I've made it with the code changes made by @jddeep. 
https://github.com/fossasia/neurolab-android/pull/426/files#diff-C1D91222ED647EF4AFC124D62E211444R60

Please confirm with him

**Comment by jddeep** [comment]
:thinking: I had kept one color code for the Analog inputs as they are of fhe same category. Okay so if they are to be recognised individually @adityastic please update with 4 different color codes. I shall update pin details in the code accordingly.

**Comment by adityastic** [comment]
@jddeep update the pr, I'll read the pr and update accordingly

**Comment by jddeep** [comment]
> @jddeep update the pr, I'll read the pr and update accordingly

Ok wait a sec

**Comment by jddeep** [comment]
@adityastic PR updated. Now you can update the colormaps :rocket:

**Comment by adityastic** [comment]
@jddeep sorry, was busy with packing. updated, have a look 🚀 
back:

front:

**Review by CloudyPadmal — APPROVED** [review]
We'll go with this

**Pull request merged**